### DISCUSSION TITLE
Hydration error

### DISCUSSION THREAD

**Discussion by AndreHaupt85** [neuefische/questions#207] [Web] [answered]
Hello all together,

i have a Error message called Hydration Error.
There was some Posts here in Discussions befor but i was not able to compare them with my problem.

Also i dont know what part of code i should post here in Discussions because i can not really imagine where the problem comes from.

![Bildschirmfoto 2022-10-04 um 13 01 07](https://user-images.githubusercontent.com/109967968/193803197-e0bebe60-92b9-4b77-9171-db41d0ff8c48.png)

This is the pull request for the problem
https://github.com/AndreHaupt85/capstone-project/pull/10

**Comment by NikRoe** [comment, marked as answer]
Hey Andre,
the issue seems to be connected to the way you work with `a`-tags in your `/pages/library.js`.

Your code currently looks like this:
```js
                     <Librarybox>
                        {recipes.map((recipe) => (
                            <Cocktail key={recipe.id}>
                                <CocktailLink href={`/details/${recipe.id}`}>
                                    <a>{recipe.name}</a>
                                </CocktailLink>
                            </Cocktail>
                        ))}
                    </Librarybox>
``` 
Your styled component `CocktailLink` is an `a`-tag. Thus, you have an `a`-tag wrapped by another `a`-tag.

You should replace your `CocktailLink` component with the `Link` component provided by nextJS and your `a`-tag with your `CocktailLink` component.

Don't forget to work with `passHref` since you're working with a custom component within the `Link` component.
For more information see this [link](https://nextjs.org/docs/api-reference/next/link).
